src/resource_clients/dataset.ts

```diff
@@ -73,14 +74,20 @@ export class DatasetClient<
             }),
         );
 
-        const response = await this.httpClient.call({
-            url: this._url('items'),
-            method: 'GET',
-            params: this._params(options),
-            timeout: DEFAULT_TIMEOUT_MILLIS,
-        });
+        const listItems = async (
+            datasetListOptions: DatasetClientListItemOptions = {},
+        ): Promise<PaginatedList<Data>> => {
+            const response = await this.httpClient.call({
+                url: this._url('items'),
+                method: 'GET',
+                params: this._params(datasetListOptions),
+                timeout: DEFAULT_TIMEOUT_MILLIS,
+            });
+
+            return this._createPaginationList(response, datasetListOptions.desc ?? false);
+        };
 
-        return this._createPaginationList(response, options.desc ?? false);
+        return this._listPaginatedFromCallback(listItems.bind(this), options);
```

Comment:
JS nit : since `const listItems()` is an arrow function, it holds `this` reference in closure (and bind is noop on it):

<img width="400" height="198" alt="image" src="https://github.com/user-attachments/assets/3ce666f5-dc37-487d-b3e8-ac99d1ca7c07" />

<img width="291" height="216" alt="image" src="https://github.com/user-attachments/assets/67034094-9987-4d6c-aad2-f9a9cc5a29f8" />

Reply:
(tldr, you can drop `.bind()` here :))